### PR TITLE
Update Security docs and properties files

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This moves the properties file config out of the core Elytron module, and also adds support for hashed passwords to remove the need for plaintext passwords in the files.

@gsmet this has a high chance to conflict so a prompt review would be appreciated.

### CODE REVIEW

**Comment by geoand** [comment]
I went through the doc update quickly and it looks good. I would personally merge this quickly assuming the tests pass so people can start looking into what is changing with security.

**Comment by geoand** [comment]
The gradle plugin test failure seems totally unrelated. I'll restart it when the rest of the tests finish.

File: core/deployment/src/main/java/io/quarkus/deployment/builditem/FeatureBuildItem.java
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/builditem/FeatureBuildItem.java]:
```suggestion
    public static final String SECURITY_PROPERTIES_FILE = "security-properties-file";
```
would be better I think. We totally removed the `elytron` part from the other extension names.

File: bom/deployment/pom.xml
Comment (gsmet) [inline bom/deployment/pom.xml]: I'm wondering if we should introduce `elytron` now? We named everything `security` before to not expose it.

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/elytron-properties-guide.adoc]: It should be `----` for sources.

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/elytron-properties-guide.adoc]: Is there any other hash available? `md5` is really not considered secured these days.

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/elytron-properties-guide.adoc]: This is not true anymore?

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/elytron-properties-guide.adoc]: The `<<subject-example>>` cross reference points to nothing existing in this document.

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/elytron-properties-guide.adoc]: I pushed a small update with some formatting improvements.

File: docs/src/main/asciidoc/keycloak-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/keycloak-guide.adoc]: should the variable be renamed too?

File: docs/src/main/asciidoc/keycloak-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/keycloak-guide.adoc]: We don't support that anymore?

File: docs/src/main/asciidoc/security-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-guide.adoc]:
```suggestion
Quarkus comes with build in security to allow for RBAC based on the common security annotations `@RolesAllowed`,
```

File: docs/src/main/asciidoc/security-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-guide.adoc]:
```suggestion
|link:oauth2-guide.html[quarkus-elytron-security-oauth2]
```

File: docs/src/main/asciidoc/security-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-guide.adoc]:
```suggestion
|Provides support for OAuth2 flows using Elytron. This extension will likely be deprecated soon and replaced by a reactive Vert.x version.
```

File: docs/src/main/asciidoc/security-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-guide.adoc]:
```suggestion
|A Microprofile JWT implementation that provides support for authenticating using Json Web Tokens. This also allows you to inject the token and claims into the application as per the MP JWT spec.
```

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/elytron-properties-guide.adoc]: I wonder if it would be better to have a `security` prefix (so `quarkus.security.users`). For users, it might be self explanatory but it's not a given it will be for everything.

File: docs/src/main/asciidoc/security-guide.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-guide.adoc]: why did we rename it `vertx-keycloak`? It doesn't work at all in a servlet environment?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a commit for doc formatting and a couple of comments.

My main concern is about the names of the extensions:
- do we really want to expose `elytron`? Until now we refrained to do it.
- is the Vert.x component of `vertx-keycloak` important for the users? My main concern is that in the vast majority of cases, people won't know they are using Vert.x as they will just have included the RESTEasy extension. Thus they might think that `vertx-keycloak` is not for them.

File: bom/deployment/pom.xml
Comment (stuartwdouglas) [reply bom/deployment/pom.xml]: The artifacts were still called Elytron. I am ok with renaming it.

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (stuartwdouglas) [reply docs/src/main/asciidoc/elytron-properties-guide.adoc]: LegacyProperyRealm in Elytron only supports this, and it's not supposed to be used in production anyway. The other advantage is that lots of systems have md5 available so you can generate the password by with echo -n user:realm:password | md5.

We could change it but it would be a bit of work as it would mean forking the Elytron implementation, and we would also need to come up with some kind of tool to enable the users to generate the passwords ( I want to do this eventually anyway, but not as an ad-hoc thing).

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (stuartwdouglas) [reply docs/src/main/asciidoc/elytron-properties-guide.adoc]: Oops, I wrote this before I decided I was going to go ahead and implement the hashed passwords

File: docs/src/main/asciidoc/keycloak-guide.adoc
Comment (stuartwdouglas) [reply docs/src/main/asciidoc/keycloak-guide.adoc]: Nope. That was specific to the Keycloak Undertow adaptor.

File: docs/src/main/asciidoc/security-guide.adoc
Comment (stuartwdouglas) [reply docs/src/main/asciidoc/security-guide.adoc]: Because our naming format is productname-feature. In this case vertx is providing Keycloak, so according to our standard naming rules this should be the name. I am happy to change it to just keycloak, but it might even need to be something more generic (as it may just be the open id connect implementation)

File: bom/deployment/pom.xml
Comment (gsmet) [reply bom/deployment/pom.xml]: Yeah, just saw that. I would say let's not block this PR for that. We can get it in and then decide to rename all of them consistently if we want to do so.

File: docs/src/main/asciidoc/elytron-properties-guide.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/elytron-properties-guide.adoc]: Yeah, probably not worth it. It's just that having `md5` in a security related documentation could look like a joke.

File: docs/src/main/asciidoc/security-guide.adoc
Comment (gsmet) [reply docs/src/main/asciidoc/security-guide.adoc]: Can we have a discussion about the naming on the mailing list? In respect to the `elytron` thing and `vertx-keycloak`? Let's not block this PR for that.

**Review by gsmet — APPROVED** [review]
Let's get this in and then we can discuss the naming globally.

**Pull request merged**
